tests/reader.rs

```diff
@@ -0,0 +1,348 @@
+use osmpbfreader::objects::{Node, NodeId, OsmId, Ref, Relation, RelationId, Tags, Way, WayId};
+use std::fs::File;
+
+#[derive(Debug, Default, PartialEq)]
+struct Stats {
+    num_nodes: i64,
+    num_ways: i64,
+    num_relations: i64,
+    sum_nodes_ids: i64,
+    sum_ways_ids: i64,
+    sum_relations_ids: i64,
+}
+
+const PBF: &str = "tests/clipperton.osm.pbf";
+const PBF_STATS_ALL: Stats = Stats {
+    num_nodes: 3097,
+    num_ways: 151,
+    num_relations: 17,
+    sum_nodes_ids: 22498900142500,
+    sum_ways_ids: 155207265285,
+    sum_relations_ids: 112821863,
+};
+const PBF_STATS_HIGHWAY: Stats = Stats {
+    num_nodes: 9,
+    num_ways: 3,
+    num_relations: 0,
+    sum_nodes_ids: 114394331047,
+    sum_ways_ids: 4118264166,
+    sum_relations_ids: 0,
+};
+
+fn check_items() -> (Vec<Node>, Vec<Way>, Vec<Relation>) {
+    let nodes: Vec<Node> = vec![
+        Node {
+            id: NodeId(1724610081),
+            tags: Tags::new(),
+            decimicro_lat: (10.2871538 * 1e7) as i32,
+            decimicro_lon: (-109.2143679 * 1e7) as i32,
+        },
+        Node {
+            id: NodeId(1724610082),
+            tags: Tags::new(),
+            decimicro_lat: (10.2872043 * 1e7) as i32,
+            decimicro_lon: (-109.2157633 * 1e7) as i32,
+        },
+        Node {
+            id: NodeId(2710971331),
+            tags: {
+                let mut tags = Tags::new();
+                tags.insert("historic".into(), "wreck".into());
+                tags.insert("seamark:type".into(), "wreck".into());
+                tags
+            },
```

Comment:
```suggestion
            tags: [("historic".into(), "wreck".into()), ("seamark:type".into(), "wreck".into())].into_iter().collect(),
```

Or somethnig similar (not tested)

Reply:
Good point, I will try this.

Reply:
It works, thanks!